### ISSUE TITLE
Don't deepfreeze on each environmental action

### ISSUE THREAD

**Issue by Bobronium** [sbidoul/hatch-pip-deepfreeze#2]
Whenever I use `hatch run` with `pip-deepfreeze` environments, it seems like it tries to perform pip install and depfreeze, regardless of whether dependencies were changed in the pyproject.toml or not.

For example, it takes almost 10 seconds just to display current python version in such env:
```console
No change to /Users/bobronium/.../requirements-all.txt
No change to /Users/bobronium/.../requirements.txt
Python 3.9.16
hatch run python --version  9.44s user 1.31s system 86% cpu 12.495 total
```
This is what I get without `type = "pip-deepfreeze"`
```console
hatch run python --version  0.51s user 0.11s system 98% cpu 0.626 total
```

Ideally, I would expect deepfreeze to run only when `hatch` decides to sync environment AND finds something to be changed.

**Comment by sbidoul** [comment]
@Bobronium thanks for your interest !

Removing the forced sync raises the question of what it means for the environment to be in sync.

If I remember correctly, hatch only checks that the dependencies are installed?

But if one uses this plugin, it is is presumably because one wants precise versions to be installed.

So what we would need is a more sophisticated implementation of `dependencies_in_sync()`, which is also not trivial.

**Comment by Bobronium** [comment]
Right, my bad. Well, I don't have a quick solution here, unfortunately.

And maybe it's worth to move our focus on implementing this in hatch, since https://github.com/pypa/pip/pull/10748, which seemingly was a blocker for this, is resolved now.
